.github/actions/split_version/action.yaml

```diff
@@ -0,0 +1,47 @@
+name: 'Split Version'
+description: 'Splits the inputed version into major, minor, and patch components'
+inputs:
+  version: 
+    description: 'Version to split'
+    required: true
+  release-branch:
+    description: 'Release branch'
+    required: true
+  existing-version:
+    description: 'Existing version'
+    required: true
+outputs:
+  minor-version:
+    description: "Minor version"
+    value: ${{ steps.split.outputs.minor-version }}
+  patch-version:
+    description: "Patch version"
+    value: ${{ steps.split.outputs.patch-version }}
+  is-pre-release:
+    description: "Is pre-release version"
+    value: ${{ steps.split.outputs.is-pre-release }}
+  release-branch:
+    description: "Is pre-release version"
+    value: ${{ steps.split.outputs.release-branch }}
+  existing-version:
+    description: "Is pre-release version"
+    value: ${{ steps.split.outputs.existing-version }}
+
+runs:
+  using: "composite"
+  steps:
+    - name: Split
+      id: split
+      shell: bash
+      run: |
+        major=$(echo ${{ inputs.version }} | cut -d "." -f 1) 
+        minor=$(echo ${{ inputs.version }} | cut -d "." -f 2) 
+        patch=$(echo ${{ inputs.version }} | cut -d "." -f 3 | cut -d "-" -f 1)  
+        minor_version=$major.$minor
+        patch_version=$major.$minor.$patch
+        echo "minor-version=$minor_version" >> $GITHUB_OUTPUT 
+        echo "patch-version=$patch_version" >> $GITHUB_OUTPUT
+        echo "is-pre-release=$( [ "${{ inputs.version }}" != "$patch_version" ] && echo true || echo false)" >> $GITHUB_OUTPUT
+        echo "release-branch=$( [ "${{ inputs.releaseBranch }}" != "" ] && echo "${{ inputs.releaseBranch }}" || echo "aissemble-release-$minor_version")" >> $GITHUB_OUTPUT
```

Comment:
Q: Have you verified this works having the `echo` and part of the `&&` conditional?

Reply:
Yes this works!